### ISSUE TITLE
sysadm generate gigabytes of logs and causes system stalls

### ISSUE THREAD

**Issue by hxw** [trueos/trueos-core#1458]
I am adding the bug here as the sysadm issues tracker is not open, please move this if necessary

---

I also noticed sysadm-binary causing high load, but seems to be more related to IO.
it was writing lots of data to its log file which now stands at:

```
% du -shc /var/log/sysadm*
643M	/var/log/sysadm                (sum of ls -l  gives: 6.2GB)
645M	/var/log/sysadm-server-ws.log  (uncompressed size is 5.7GB)
1.3G	total                          (compressed size)

```
This whole sysadm log appears to be multiple copies of /var/log/lpreserver/lpreserver.log
with some quotes added. (around 1500 copies using a quick grep/wc -l)

Since the lpreserver does not appear to rotate its log either this is making the problem worse
especially since I have hourly snapshots. (I had been thinking the system freezing for about 5…10min every hour was just the ZFS replication)

So far I have not found any documentation to rotate this log file; so will manually mv and touch new logs (both the sysadm and lpreserver) and check later to see if the new files are used (no programs were restarted or signalled for this test)

---

After running overnight:

* renamed log: sysadm-server-ws.log.1 has increased i.e. sysadm-binary does not re-open the file.
* lpreserver does use the rotated file.

Also in the sysadm/websocket dir there are many weekly/daily logs: e.g.

```
-rw-r–r-- 1 root wheel 51413700 2017-07-26 23:01 events-lifepreserver-2017-07-26.log
-rw-r–r-- 1 root wheel   263674 2017-07-27 09:01 events-lifepreserver-2017-07-27.log

```
from the above two I can see that rotating lpreserver.log reduced that log by a couple of orders of magnitude.

Restarting sysadm simply renames the log to .old and does not do a proper rotation; but system is fine and there was no noticable slowdown during the last lpreserver replication.

## Current workarounds
* rotate lpreserver log using newsyslog
* service sysadm restart daily

**Comment by hxw** [comment]
~~~
# cat /etc/newsyslog.conf.d/lpreserver.conf 
# lpreserver.conf
/var/log/lpreserver/lastrep-recv-log		644  7	   1000	*     J
/var/log/lpreserver/lastrep-send-log		644  7	   1000	*     J
/var/log/lpreserver/lpreserver.log		644  7	   1000	*     J
/var/log/lpreserver/lpreserver_failed.log	644  7	   1000	*     J
~~~

**Comment by beanpole135** [comment]
I just tracked this down and found a couple debugging statements that were still enabled for the Life Preserver log parser that was causing the main "ws-server.log" file to get too large. This should prevent that log file from getting large again, since it is primarily used for debugging and some general server-status reporting (automatically rotated on server stop/start - and almost nothing should be getting put there anymore).

I also changed the default "pruner" for the log directory to start pruning logs older than 30 days by default instead of 90 days.
This can be individually customized with the "prune_log_days_old" setting in /var/db/sysadm.ini as follows:
```
[General]
prune_log_days_old=30
```
